code/game/machinery/Sleeper.dm

```diff
@@ -119,30 +119,37 @@
 /obj/machinery/sleep_console/Topic(href, href_list)
 	if(..())
 		return
-	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
-		usr.set_interaction(src)
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr
+	if ((user.contents.Find(src) || ((get_dist(src, user) <= 1) && isturf(loc))))
+		user.set_interaction(src)
 		if (href_list["chemical"])
-			if (src.connected)
-				if (src.connected.occupant)
-					if (src.connected.occupant.stat == DEAD)
-						to_chat(usr, "\red \b This person has no life for to preserve anymore. Take them to a department capable of reanimating them.")
-					else if(src.connected.occupant.health > 0 || href_list["chemical"] == "inaprovaline")
-						src.connected.inject_chemical(usr,href_list["chemical"],text2num(href_list["amount"]))
+			if (connected)
+				if (connected.occupant)
```

Comment:
Try to reduce indentation and save on white space and combine your conditionals if it doesn't make it confusing